airdrop/application/services/airdrop_services.py

```diff
@@ -130,14 +123,73 @@ def get_airdrop_window_stake_details(self, inputs):
 
         return status, response
 
-    def get_stake_info(self, staking_contract_address, address):
-        contract = get_contract_instance(
-            STAKING_CONTRACT_PATH, staking_contract_address, contract_name='STAKING')
+    def get_stake_info(self, contract_address, address, airdrop_rewards):
+        try:
+
+            is_stake_window_is_open, max_stake_amount = self.get_stake_window_details(
+                contract_address)
+
+            is_user_can_stake, already_staked_amount = self.get_stake_details_of_address(
+                contract_address, address)
+
+            is_stakable, stakable_amount, tranfer_to_wallet = self.get_stake_and_claimable_amounts(
+                airdrop_rewards, is_stake_window_is_open, max_stake_amount, is_user_can_stake, already_staked_amount)
+
+            return is_stakable, stakable_amount, tranfer_to_wallet
+        except BaseException as e:
+            raise e("Exception on get_stake_info {}".format(e))
+
+    def get_stake_and_claimable_amounts(self, airdrop_rewards, is_stake_window_is_open, max_stake_amount, is_user_can_stake, already_staked_amount):
+        # User can stake if stake window is open and user can stake
+        is_stakable = True if is_user_can_stake and is_stake_window_is_open else False
+
+        allowed_amount_for_stake = max_stake_amount - already_staked_amount
+
+        if(airdrop_rewards <= allowed_amount_for_stake):
+            # If airdrop rewards is less than allowed amount for stake then stake the full airdrop rewards
+            stakable_amount = airdrop_rewards
+        else:
+            stakable_amount = allowed_amount_for_stake
```

Comment:
when airdropw_rewards > allowed amount_for_stake 
=> stakable_amount = airdrop_rewards - allowed_amount_for_stake , Please check this agains the logic in our test case

Reply:
please have a test case showing all the scenarios 
-  Full rewaards staked
-  Partial rewards staked
-  full rewards sent back to the user wallet and nothing could be staked as the user exceeded all the limit 
-  full rewards to wallet as the stake window is not open

Reply:
Updated the test case in [commit](https://github.com/singnet/airdrop-services/pull/94/commits/387f89de55b5762ca4b37cfdc8aaef360afee03e)